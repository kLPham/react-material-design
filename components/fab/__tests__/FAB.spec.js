import React from 'react';
import { shallow } from 'enzyme';
import FAB from '../index';

describe('<FAB />', () => {
    it('should render and have span element', () => {
        const icon = 'bug_report';
        const fab = shallow(
            <FAB icon={icon} />,
            );
        expect(fab).toMatchSnapshot();
    });
    it('should render mini', () => {
        const mini = true;
        const icon = 'bug_report';
        const fab = shallow(
            <FAB icon={icon} mini={mini} />,
        );
        expect(fab).toMatchSnapshot();
    });
    it('should render plain', () => {
        const plain = true;
        const icon = 'bug_report';
        const fab = shallow(
            <FAB icon={icon} plain={plain} />,
        );
        expect(fab).toMatchSnapshot();
    });
    it('should render absolutely positioned', () => {
        const absolute = true;
        const icon = 'bug_report';
        const location = 'absolute-bottom-right';
        const fab = shallow(
            <FAB icon={icon} absolute={absolute} location={location} />,
        );
        expect(fab).toMatchSnapshot();
    });
    it('should render floating', () => {
        // TODO: should render floating
    });
});
