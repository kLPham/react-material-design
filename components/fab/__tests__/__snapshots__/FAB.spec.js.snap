// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FAB /> should render absolutely positioned 1`] = `
<button
  aria-label="bug_report"
  className="mdc-fab app-fab--absolute material-icons"
  style={
    Object {
      "bottom": "1rem",
      "position": "absolute",
      "right": "1rem",
    }
  }
>
  <span
    className="mdc-fab__icon"
  >
    bug_report
  </span>
</button>
`;

exports[`<FAB /> should render and have span element 1`] = `
<button
  aria-label="bug_report"
  className="mdc-fab  material-icons"
>
  <span
    className="mdc-fab__icon"
  >
    bug_report
  </span>
</button>
`;

exports[`<FAB /> should render mini 1`] = `
<button
  aria-label="bug_report"
  className="mdc-fab mdc-fab--mini material-icons"
>
  <span
    className="mdc-fab__icon"
  >
    bug_report
  </span>
</button>
`;

exports[`<FAB /> should render plain 1`] = `
<button
  aria-label="bug_report"
  className="mdc-fab mdc-fab--plain material-icons"
>
  <span
    className="mdc-fab__icon"
  >
    bug_report
  </span>
</button>
`;
